### PR TITLE
SC_fftlib: allow ensurewindow to be called

### PR DESCRIPTION
**Pull request by sonoro1234**
this allow growing fftsize correcting bug that avoids it

### CODE REVIEW

**Comment by sonoro1234** [comment]
more info in:
http://new-supercollider-mailing-lists-forums-use-these.2681727.n2.nabble.com/SC-FFT-MAXSIZE-32768-td7625039.html

**Comment by danstowell** [comment]
The original intention was: if the initial max size is 2^9, and someone requests 2^17, we simply initialise the single FFT at 2^17. Why would you also initialise (2^10 .. 2^16) as well? It seems that's what you've gone for.

**Comment by sonoro1234** [comment]
If we only initialize 2^17 and afterwards 2^16 this second one would not be detected because it is less than largest_log2n (wich has been updated to 2^17 in first initialization)

**Comment by sonoro1234** [comment]
Another option would be to check precompiledForwardPlansInPlace[f->log2nfull] and if it is null call ensurewindow that would only make plans and window for that size and get rid of largest_log2n.
What do you think?

**Comment by danstowell** [comment]
Ah, I see. There may have been a logic error in the previous implementation then! Yes, I think the null check makes sense...

**Comment by sonoro1234** [comment]
On second thought It would break SC_FFT_VDSP and other fft implementations because this check only works on FFTW

**Comment by danstowell** [comment]
Check fftWindow[] for null? (Problematic for rect window unfortunately.............)

**Comment by sonoro1234** [comment]
Yes rect window breaks this solution.
Beasides I dont see allocating intermediate sizes as a big problem, It is just as if we had begun with a bigger SC_FFT_MASIZE (memory waste)

**Comment by sonoro1234** [comment]
Perhaps would make more sense reduce SC_FFT_ABSOLUTE_MAXSIZE (which is clearly too big)

File: common/SC_fftlib.cpp [hunk 1]
```diff
@@ -271,8 +273,10 @@ scfft * scfft_create(size_t fullsize, size_t winsize, SCFFT_WindowFunction winty
 	f->trbuf   = trbuf;
 
 	// Buffer is larger than the range of sizes we provide for at startup; we can get ready just-in-time though
-	if (fullsize > SC_FFT_MAXSIZE)
+	if (fullsize > largest_fftsize){
+		//printf("calling scfft_ensurewindow\n");
```
Comment (danstowell) [inline common/SC_fftlib.cpp]: probably best to remove comment cruft

**Comment by danstowell** [comment]
Yes, could do.

OK I'm basically happy for this to be merged - I haven't tested it at all though; would be a good idea for someone to confirm that a few FFT-based synthdefs (from the helpfiles) continue to work...

**Comment by sonoro1234** [comment]
which could be the best SC_FFT_ABSOLUTE_MAXSIZE ?

**Comment by sonoro1234** [comment]
264144 seems enough.

**Comment by sonoro1234** [comment]
11 days waiting for test means that this can get lost in the list of old PRs.
In this one and #2015 there was an obvious bug (waiting for 3 years in PartConv issue #734) and at least it seems corrected.
I would merge since it wont be worse than the code we had.

**Comment by timblechmann** [comment]
looks dangerous:

```
/home/travis/build/timblechmann/supercollider/common/SC_fftlib.cpp:86:0: warning: "SC_FFT_ABSOLUTE_MAXSIZE" redefined
 #define SC_FFT_ABSOLUTE_MAXSIZE 2147483648
 ^
In file included from /home/travis/build/timblechmann/supercollider/common/SC_fftlib.cpp:35:0:
/home/travis/build/timblechmann/supercollider/include/common/SC_fftlib.h:35:0: note: this is the location of the previous definition
 #define SC_FFT_ABSOLUTE_MAXSIZE 262144
 ^
/home/travis/build/timblechmann/supercollider/common/SC_fftlib.cpp:87:0: warning: "SC_FFT_LOG2_ABSOLUTE_MAXSIZE" redefined
 #define SC_FFT_LOG2_ABSOLUTE_MAXSIZE 31
 ^
In file included from /home/travis/build/timblechmann/supercollider/common/SC_fftlib.cpp:35:0:
/home/travis/build/timblechmann/supercollider/include/common/SC_fftlib.h:36:0: note: this is the location of the previous definition
 #define SC_FFT_LOG2_ABSOLUTE_MAXSIZE 18
 ^
/home/travis/build/timblechmann/supercollider/common/SC_fftlib.cpp:88:0: warning: "SC_FFT_LOG2_ABSOLUTE_MAXSIZE_PLUS1" redefined
 #define SC_FFT_LOG2_ABSOLUTE_MAXSIZE_PLUS1 32
 ^
In file included from /home/travis/build/timblechmann/supercollider/common/SC_fftlib.cpp:35:0:
/home/travis/build/timblechmann/supercollider/include/common/SC_fftlib.h:37:0: note: this is the location of the previous definition
 #define SC_FFT_LOG2_ABSOLUTE_MAXSIZE_PLUS1 19
```

**Comment by sonoro1234** [comment]
@timblechmann yes it is. I was not aware of defines present in both places. Solution is to delete those present in SC_fftlib.cpp, 
my fork is broken now , could you please do it?

**Comment by sonoro1234** [comment]
@timblechmann Dont mind, I am repairing my fork and preparing a PR

**Comment by sonoro1234** [comment]
corrected in #2089

**Pull request merged**